app/src/webkit/java/org/mozilla/focus/webkit/WebkitView.java

```diff
@@ -189,6 +190,11 @@ public void reload() {
         }
     }
 
+    @Override
+    public @SiteIdentity.SecurityState int getSecurityState() {
+        return getCertificate() == null ? SiteIdentity.INSECURE : SiteIdentity.SECURE;
+    }
+
```

Comment:
I am not pretty sure whether this logic is enough. Maybe we can add an issue to track this.